File: two_sum/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const obj = {};
    for (let i=0; i<nums.length; i++){
        obj[nums[i]]= i;
    }
    for (let i=0; i<nums.length; i++){
        if(obj[target-nums[i]]){
            return [i,obj[target-nums[i]]];
        }
    }
};

File: two_sum/two_sum.rb
# Given an array of integers, return indices of the two numbers such that they add up to a specific target.
#
# You may assume that each input would have exactly one solution, and you may not use the same element twice.
#
# Example:
#
# Given nums = [2, 7, 11, 15], target = 9,
#
# Because nums[0] + nums[1] = 2 + 7 = 9,
# return [0, 1].


# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    hash = Hash.new(false);
    nums.each_with_index do |int, i|
        hash[int]= i
    end
    nums.each_with_index do |int, i|
       return [i,hash[target-int]] if hash[target-int]
    end
end
